ebean-core/src/main/java/io/ebeaninternal/server/query/SqlTreeBuilder.java

```diff
@@ -172,7 +172,7 @@ static String mergeOnDistinct(String idCols, String dbOrderBy) {
     String[] split = idCols.split(",");
     for (String col : split) {
       col = col.trim();
-      if (!dbOrderBy.contains(col)) {
+      if (!DbOrderByTrim.contains(dbOrderBy, col)) {
```

Comment:
I think, this was not so good before. What if dbOrderBy = `t0.name_long` and col = `t0.name`

Reply:
Agreed.